test/jdk/java/lang/instrument/NativeMethodPrefixApp.java

```diff
@@ -22,25 +22,160 @@
+        if (testSrc == null) {
+            throw new Exception("test.src system property isn't set");
+        }
+        final Path bootReporterSourceDir = Path.of(testSrc)
+                .resolve("bootreporter").toAbsolutePath();
+        // this directory must be present
+        if (!Files.isDirectory(bootReporterSourceDir)) {
+            throw new Exception(bootReporterSourceDir.toString() + " is missing or not a directory");
+        }
+        final String[] javacCmdArgs = {"-d", destDir.toString(),
+                bootReporterSourceDir.resolve("StringIdCallback.java").toString(),
+                bootReporterSourceDir.resolve("StringIdCallbackReporter.java").toString()
+        };
+        System.out.println("invoking javac with args: " + Arrays.toString(javacCmdArgs));
+        final int exitCode = JAVAC_TOOL.run(System.out, System.err, javacCmdArgs);
+        if (exitCode != 0 ){
+            throw new Exception("javac command failed with exit code: " + exitCode);
+        }
+    }
+
+    private static void compileAgentClasses(final Path destDir) throws Exception {
+        // directory containing this current test file
+        final String testSrc = System.getProperty("test.src");
+        if (testSrc == null) {
+            throw new Exception("test.src system property isn't set");
+        }
+        final Path agentJavaSrc = Path.of(testSrc).resolve("NativeMethodPrefixAgent.java")
+                .toAbsolutePath();
+        // this file must be present
+        if (!Files.isRegularFile(agentJavaSrc)) {
+            throw new Exception(agentJavaSrc.toString() + " is missing or not a file");
+        }
+        final Path instrumentorJavaSrc = Path.of(testSrc)
+                .resolve("asmlib", "Instrumentor.java").toAbsolutePath();
+        // this file must be present
+        if (!Files.isRegularFile(instrumentorJavaSrc)) {
+            throw new Exception(instrumentorJavaSrc.toString() + " is missing or not a file");
+        }
+        final String currentRuntimeVersion = String.valueOf(Runtime.version().feature());
+        final String[] javacCmdArgs = {"-d", destDir.toString(),
+                // enable preview since asmlib/Instrumentor.java uses ClassFile API PreviewFeature
+                "--enable-preview", "--release", currentRuntimeVersion,
+                agentJavaSrc.toString(), instrumentorJavaSrc.toString()};
+        System.out.println("invoking javac with args: " + Arrays.toString(javacCmdArgs));
+        final int exitCode = JAVAC_TOOL.run(System.out, System.err, javacCmdArgs);
+        if (exitCode != 0 ){
+            throw new Exception("javac command failed with exit code: " + exitCode);
+        }
+    }
 
-    public static void main(String args[]) throws Exception {
-        (new NativeMethodPrefixApp()).run(args, System.err);
+    private static void launchApp(final Path agentJar) throws Exception {
+        final OutputAnalyzer oa = ProcessTools.executeLimitedTestJava(
```

Comment:
Is there any reason the test uses 'executeLimitedTestJava' and not 'executeTestJava'? 
It might make sense to run it with different VM flags.

Reply:
Hello Leonid, I had intentionally used `executeLimitedTestJava()` to avoid passing along any additional options when launching that test application jar because it wasn't clear to me if passing those along will be of any use. Especially, options like `-Xcomp` which might be used when launching jtreg. That was the sole reason. 

Having said that, I have now updated the tests to use `executeTestJava()` to allow them to pass along these options to the jar being launched, since prior to this change, the test would have anyway ended up using those additional jtreg launch specific options.